lib/nerves_hub/device_link.ex

```diff
@@ -130,7 +130,7 @@ defmodule NervesHub.DeviceLink do
     :ok
   end
 
-  defp maybe_clear_inflight_update(_device, %{"currently_downloading_uuid" => _uuid}), do: :ok
+  defp maybe_clear_inflight_update(_device, %{"currently_downloading_uuid" => uuid}) when not is_nil(uuid), do: :ok
```

Comment:
```suggestion
  defp maybe_clear_inflight_update(_device, %{"currently_downloading_uuid" => uuid}) when is_binary(uuid) and byte_size(uuid) == 36, do: :ok
```
You could also maybe flip this so it is the expected success case? (Could also be `byte_size > 0`)

```elixir
iex(1)> byte_size "df1b7b8f-66f8-5f1a-09a6-1c84d0b69126"
36
```